### DISCUSSION TITLE
IDP - Authentication token with user session note

### DISCUSSION THREAD

**Discussion by mrharsalla** [keycloak/keycloak#36175] [Q&A]
Hi,
How to include user session note in authentication token after successful authentication.
Thanks

**Comment by danielFesenmeyer** [comment]
What about using the "User Session Note" [protocol mapper](https://www.keycloak.org/docs/latest/server_admin/#_protocol-mappers)?

**Reply by danielFesenmeyer** [reply]
Also see this section about "Available Session Data": https://www.keycloak.org/docs/latest/server_admin/#available-user-session-data

**Reply by mrharsalla** [reply]
Hello Daniel,

With authentication from an identity provider, I receive data from the user but I don't want to save everything in keycloak. 
Example:
- I want to save the first name, last name and email address in keycloak
I don't want to save either the date of birth or the address

But, I want keycloak to generate a token including first name, last name, email address, date of birth and address.
I tried to use the "User session note" attributes but without success.

Thanks

**Reply by danielFesenmeyer** [reply]
It seems you can use the feature implemented here: https://github.com/keycloak/keycloak/issues/17659

For both date of birth and address, you could configure these mappers:

- on IDP: "Claim to User Session Note": Map the corresponding IDP claim to a user session note X
- on protocol level (e.g. client): Map the user session note X to a keycloak token claim
